bigint.go

```diff
@@ -326,6 +326,9 @@ func addInline(xVal, yVal uint64, xNeg, yNeg bool) (zVal uint64, zNeg, ok bool)
 func mulInline(xVal, yVal uint64, xNeg, yNeg bool) (zVal uint64, zNeg, ok bool) {
 	hi, lo := bits.Mul64(xVal, yVal)
 	neg := xNeg != yNeg
```

Comment:
nit: This could be shortened to `neg := xNeg != yNeg && lo != 0` (and same with a couple others).